src/error.jl

```diff
@@ -90,15 +90,40 @@ function Base.showerror(io::IO, e::ErrorConstrDimMismatch)
 end
 
 struct InfiniteDimensionError <: Exception
+  msg::String
   check_available::Bool
-  InfiniteDimensionError(; check_available::Bool = false) = new(check_available)
+
+  @doc """
+      InfiniteDimensionError(msg; check_available = false)
+
+  A computation cannot be carried out because of an infinite-dimensional vector
+  space. `msg` is a descriptive error message.
```

Comment:
With this change, we could re-use the error also for things which are not (explicitly) vector spaces. So I'd suggest to avoid mentioning them here.

Also, at this point the computation was already attempted and we gave up, so I'd change "cannot" to "could not".
```suggestion
  A computation could not be carried out because of something being
  infinite-dimensional. `msg` is a descriptive error message.
```

Reply:
Okay. If we generalize this from vector spaces, I will also change the default message. (I would like to change/generalize this `check_available` thing too, but that would break OSCAR...)